bucket_manifest/bucket_manifest_job.py

```diff
@@ -0,0 +1,216 @@
+"""
+Module for submitting jobs to job queue and consuming SQS to generate a bucket manifest
+"""
+import os
+import sys
+import time
+from datetime import datetime
+import json
+from functools import partial
+import argparse
+import logging
+from multiprocessing.pool import Pool
+
+from urllib.parse import urlparse
+import boto3
+from botocore.exceptions import ClientError
+
+import utils
+
+logging.basicConfig(level=logging.INFO)
+logging.getLogger().addHandler(logging.StreamHandler(sys.stdout))
+
+NUMBER_OF_THREADS = 16
+MAX_RETRIES = 10
+
+REGION = os.environ.get("REGION", "us-east-1")
+
+
+def purge_queue(queue_url):
+    """
+    Remove all messages in the queue
+    """
+    client = boto3.client("sqs", region_name=REGION)
+    try:
+        client.purge_queue(QueueUrl=queue_url)
+    except client.exceptions.QueueDoesNotExist:
+        logging.error("Queue {} does not exist".format(queue_url))
+        sys.exit(1)
+
+
+def submit_job(job_queue, job_definition, key):
+    """
+    Submit job to the job queue
+
+    Args:
+        job_queue(str): job queue name
+        job_definition(str): job difinition name
+        key(str): S3 object key
+    
+    Returns:
+        None
+    """
+    client = boto3.client("batch", region_name=REGION)
+    n_tries = 0
+
+    while n_tries < MAX_RETRIES:
+        try:
+            client.submit_job(
+                jobName="bucket_manifest",
+                jobQueue=job_queue,
+                jobDefinition=job_definition,
+                containerOverrides={"environment": [{"value": key, "name": "KEY"}]},
+            )
+            logging.info("submitting job to compute metadata of {}".format(key))
+            break
```

Comment:
I think you could just remove this break and indent the sleep later on so it's in the except block

Reply:
The `break` is needed to break the while loop. If `submit_job` is success just break the loop